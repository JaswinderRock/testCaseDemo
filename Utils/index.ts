export const findByTestAttr = (component: any, attr: String) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
}
export const findByClassName = (component: any, className: String) => {
    const wrapper = component.find(`.${className}`);
    return wrapper;
}