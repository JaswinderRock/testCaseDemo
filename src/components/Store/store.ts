import { createStore } from 'redux';
import { cakeReducer } from '../Reducer/cakeReducer';
import reducer from '../Reducer/incrementReducer';

export const incrementStore = createStore(reducer)
export const Store = createStore(cakeReducer);