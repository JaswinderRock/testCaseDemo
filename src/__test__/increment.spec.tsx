import React from 'react';
import { mount } from "enzyme"
import { debug } from 'console';
import IncrementDecrement from '../components/Screen/Increment';
import { Provider } from 'react-redux';
import { Store } from '../components/Store/store';
import { findByTestAttr } from '../../Utils';

const setUp = () => {
    const component = mount(<Provider store={Store}>
        <IncrementDecrement />
    </Provider>);
    return component;
}
describe("Render Increment", () => {
    let component: any;

    beforeEach(() => {
        component = setUp();
    });
    test("increment test is render  ", () => {
        const component = setUp();
        debug(component)
        expect(component).toMatchSnapshot()
    })
    test("renders correctly with initial count", () => {
        const wrapper = setUp();
        expect(wrapper.find("h1").text()).toContain("Count: 0");
    })
    it('calls the increment function when the "Increment" button is clicked', () => {
        const incrementButton = findByTestAttr(component, 'increment-btn');
        expect(incrementButton.exists()).toBe(true);
        incrementButton.simulate('click');
        const wrapper = setUp();
        expect(wrapper.find("h1").text()).toContain("Count: 1");
    });
    it('calls the decrement function when the "Decrement" button is clicked', () => {
        const decrementButton = findByTestAttr(component, 'decrement-btn')
        expect(decrementButton.exists()).toBe(true);
        decrementButton.simulate('click');
        const wrapper = setUp();
        expect(wrapper.find("h1").text()).toContain("Count: 0");
    });
})