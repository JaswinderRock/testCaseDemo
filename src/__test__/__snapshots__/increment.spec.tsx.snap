// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Increment increment test is render   1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(IncrementDecrement)>
    <IncrementDecrement
      count={0}
      decrement={[Function]}
      increment={[Function]}
    >
      <div
        className="IncrementDecrement"
      >
        <h1
          className="Count-Text"
          data-test="Count-Text"
        >
          Count: 
          0
        </h1>
        <button
          className="Increment-btn"
          data-test="increment-btn"
          onClick={[Function]}
          style={
            Object {
              "background": "blue",
              "borderRadius": "4px",
              "color": "white",
              "fontSize": "18px",
              "padding": "8px",
            }
          }
          type="button"
        >
          Increment
        </button>
        <button
          className="decrement-btn"
          data-test="decrement-btn"
          onClick={[Function]}
          style={
            Object {
              "background": "blue",
              "borderRadius": "4px",
              "color": "white",
              "fontSize": "18px",
              "marginLeft": "10px",
              "padding": "8px",
            }
          }
        >
          Decrement
        </button>
      </div>
    </IncrementDecrement>
  </Connect(IncrementDecrement)>
</Provider>
`;
